File: src/wrappers/python/examples/rest_example.py
#!/usr/bin/env python3
"""
ITSCAM SDK - REST API example (Python)

Logs into a camera over HTTP (or HTTPS with --https) and fetches a few
config endpoints to demonstrate the wrapper.

Usage::

    python3 rest_example.py <host> <user> <password> [--https]

Copyright (c) 2026 Pumatronix
"""
from __future__ import annotations

import argparse
import json
import sys

from itscam import ItscamRestClient


def main() -> int:
    p = argparse.ArgumentParser(description=__doc__)
    p.add_argument("host")
    p.add_argument("user")
    p.add_argument("password")
    p.add_argument("--https", action="store_true",
                   help="use HTTPS instead of HTTP")
    p.add_argument("--insecure", action="store_true",
                   help="skip TLS server certificate verification")
    args = p.parse_args()

    scheme = "https" if args.https else "http"
    port   = 443 if args.https else 80

    with ItscamRestClient() as rest:
        rest.set_base_url(args.host, port, scheme)
        if args.https and args.insecure:
            rest.set_verify_server_certificate(False)
        login_resp = rest.login(args.user, args.password)
        print("login OK:", json.dumps(login_resp, indent=2))

        info = rest.get("/api/equipment/misc/readonly/volatile")
        print("\nvolatile info:", json.dumps(info, indent=2))

        profiles = rest.get("/api/image/profiles")
        print("\nprofiles:", json.dumps(profiles, indent=2))

    return 0


if __name__ == "__main__":
    sys.exit(main())

File: src/wrappers/nodejs/examples/rest-example.js
#!/usr/bin/env node
/*
 * SPDX-License-Identifier: Proprietary
 * Copyright (c) 2026 Pumatronix
 *
 * REST API example.  Mirrors rest_example.py / rest_example.go.
 *
 * Usage:
 *   node rest-example.js <host> <user> <password> [--https] [--insecure]
 */
'use strict';

const path = require('path');
const { ItscamRestClient } = require(path.join(__dirname, '..', 'src'));

function main() {
    const args = process.argv.slice(2);
    if (args.length < 3) {
        console.error(
            'Usage: rest-example.js <host> <user> <password> [--https] [--insecure]');
        process.exit(1);
    }
    const [host, user, password, ...flags] = args;
    const useHttps = flags.includes('--https');
    const insecure = flags.includes('--insecure');
    const scheme = useHttps ? 'https' : 'http';
    const port   = useHttps ? 443 : 80;

    const rest = new ItscamRestClient();
    try {
        rest.setBaseUrl(host, port, scheme);
        if (useHttps && insecure) {
            rest.setVerifyServerCertificate(false);
        }

        const loginResp = rest.login(user, password, 10000);
        console.log('login OK:', JSON.stringify(loginResp, null, 2));

        const volatileInfo = rest.get(
            '/api/equipment/misc/readonly/volatile', 10000);
        console.log('\nvolatile info:', JSON.stringify(volatileInfo, null, 2));

        const profiles = rest.get('/api/image/profiles', 10000);
        console.log('\nprofiles:', JSON.stringify(profiles, null, 2));
    } finally {
        rest.close();
    }
}

main();
